Sources/XMTP/Client.swift

```diff
@@ -38,7 +38,27 @@ class Client {
 			// swiftlint:enable no_optional_try
 			return keys
 		} else {
-			return try await PrivateKeyBundleV1.generate(wallet: account)
+			let keys = try await PrivateKeyBundleV1.generate(wallet: account)
+
+			let keyBundle = PrivateKeyBundle(v1: keys)
+			let encryptedKeys = try await keyBundle.encrypted(with: account)
+
+			var authorizedIdentity = AuthorizedIdentity(privateKeyBundleV1: keys)
+			authorizedIdentity.address = account.address
+			let authToken = try await authorizedIdentity.createAuthToken()
+
+			var apiClient = apiClient
+			apiClient.setAuthToken(authToken)
+
+			do {
+				try await apiClient.publish(envelopes: [
+					Envelope(topic: .userPrivateStoreKeyBundle(account.address), timestamp: Date(), message: try encryptedKeys.serializedData()),
+				])
+			} catch {
+				// Unable to save keys
```

Comment:
Maybe worth bubbling this up or returning nil if it breaks things? Or can we move on without it?

Reply:
Ya, I agree. I've removed this try/catch.